### PR TITLE
Rpc analyzers sort order

### PR DESCRIPTION
**Pull request by j3parker**

### CODE REVIEW

**Comment by j3parker** [comment]
diff is awful, sorry

File: src/D2L.CodeStyle.Analyzers/RpcDependencies/RpcDependencyAnalyzer.cs [hunk 1]
```diff
@@ -18,8 +18,18 @@ internal sealed class RpcDependencyAnalyzer : DiagnosticAnalyzer {
 			description: "RPCs must take an IRpcContext, IRpcPostContext or IRpcPostContextBase as their first argument"
 		);
 
+		internal static readonly DiagnosticDescriptor SortRule = new DiagnosticDescriptor(
+			id: "D2L0005",
+			title: "Dependency-injected arguments must be properly sorted",
```
Comment (saquibmian) [inline src/D2L.CodeStyle.Analyzers/RpcDependencies/RpcDependencyAnalyzer.cs]: Dependency-injected arguments _in RPC methods_ must be properly sorted...
(just 'cause the title should be enough to disambiguate the error among diagnostics)

File: src/D2L.CodeStyle.Analyzers/RpcDependencies/RpcDependencyAnalyzer.cs [hunk 2]
```diff
@@ -18,8 +18,18 @@ internal sealed class RpcDependencyAnalyzer : DiagnosticAnalyzer {
 			description: "RPCs must take an IRpcContext, IRpcPostContext or IRpcPostContextBase as their first argument"
 		);
 
+		internal static readonly DiagnosticDescriptor SortRule = new DiagnosticDescriptor(
+			id: "D2L0005",
+			title: "Dependency-injected arguments must be properly sorted",
+			messageFormat: "Dependency-injected must come before RPC parameters but after IRpcContext",
```
Comment (saquibmian) [inline src/D2L.CodeStyle.Analyzers/RpcDependencies/RpcDependencyAnalyzer.cs]: nit: Dependency-injected _arguments_ ...
(same with description below)

File: src/D2L.CodeStyle.Analyzers/RpcDependencies/RpcDependencyAnalyzer.cs [hunk 3]
```diff
@@ -69,58 +84,108 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
...
+		) {
 			bool isRpc = method
 				.AttributeLists
 				.SelectMany( al => al.Attributes )
-				.Any( attr => IsRpcAttribute( rpcAttributeType, attr, context.SemanticModel ) );
+				.Any( attr => IsAttribute( rpcAttributeType, attr, context.SemanticModel ) );
 
 			if( !isRpc ) {
-				return;
+				return false;
```
Comment (saquibmian) [inline src/D2L.CodeStyle.Analyzers/RpcDependencies/RpcDependencyAnalyzer.cs]: I feel like this should be in `CheckThatMethodIsRpcMethod`, that is called before all of the other `CheckThat*` methods.

File: src/D2L.CodeStyle.Analyzers/RpcDependencies/RpcDependencyAnalyzer.cs (hunk 1 above)
Comment (j3parker) [inline src/D2L.CodeStyle.Analyzers/RpcDependencies/RpcDependencyAnalyzer.cs]: Excellent point

File: src/D2L.CodeStyle.Analyzers/RpcDependencies/RpcDependencyAnalyzer.cs (hunk 3 above)
Comment (j3parker) [inline src/D2L.CodeStyle.Analyzers/RpcDependencies/RpcDependencyAnalyzer.cs]: done, thanks

File: src/D2L.CodeStyle.Analyzers/RpcDependencies/RpcDependencyAnalyzer.cs (hunk 1 above)
Comment (j3parker) [inline src/D2L.CodeStyle.Analyzers/RpcDependencies/RpcDependencyAnalyzer.cs]: done

**Pull request merged**